### ISSUE TITLE
Cannot rethrow another error

### ISSUE THREAD

**Issue by QJonny** [mappum/watt#9]
Hi, I'm having problems rethrowing another kind of error inside a catch statement, when using watt.
Here is my code (note that the function is located inside a class, the method having been initialized with wrappAll):

``` javascript
* launch (next) {
    try {
      yield anAsyncMethod (someParameters, next);
    }
    catch (err) { // Here err has structure {code:'...', message:'...}
      throw 'new error';
    }
  }
```

The function calling `launch` receives the original exception instead of `new error`.  

Is there a way you can fix it? Thanks!

**Comment by Skywalker13** [comment]
A complete script in order to reproduce the problem:

``` js
'use strict';

const watt = require ('watt');

const a = watt (function * () {
  throw 'a';
  yield true;
});

const b = watt (function * () {
  try {
    yield a ();
  } catch (ex) {
    throw 'a and b';
  }
});

try {
  b ();
} catch (ex) {
  // print 'a' but 'a and b' is expected
  console.log (ex);
}
```

**Comment by Skywalker13** [comment]
ping @mappum

**Comment by mappum** [comment]
Hey, thanks for reporting this and for the nice example. I encountered this
as well a while ago, but on my first try I couldn't find the cause. I'll
investigate more deeply.

On Mon, Jul 4, 2016 at 12:26 AM, Mathieu Schroeter <notifications@github.com

**Comment by Skywalker13** [comment]
OK, thank you
I tried to fix but still without success.

**Comment by mappum** [comment]
This should now be solved as of `watt@3.3.0`. Thanks @cshum!

**Comment by Skywalker13** [comment]
Great!
